src/reporting/Market.sol

```diff
@@ -0,0 +1,604 @@
+        return reportingWindow;
+    }
+
+    function getBranch() public constant returns (Branch) {
+        return reportingWindow.getBranch();
+    }
+
+    function getAutomatedReporterDisputeBondToken() public constant returns (DisputeBondToken) {
+        return automatedReporterDisputeBondToken;
+    }
+
+    function getLimitedReportersDisputeBondToken() public constant returns (DisputeBondToken) {
+        return limitedReportersDisputeBondToken;
+    }
+
+    function getAllReportersDisputeBondToken() public constant returns (DisputeBondToken) {
+        return allReportersDisputeBondToken;
+    }
+
+    function getNumberOfOutcomes() public constant returns (uint8) {
+        return numOutcomes;
+    }
+
+    function getEndTime() public constant returns (uint256) {
+        return endTime;
+    }
+
+    function getTentativeWinningPayoutDistributionHash() public constant returns (bytes32) {
+        return tentativeWinningPayoutDistributionHash;
+    }
+
+    function getFinalWinningReportingToken() public constant returns (ReportingToken) {
+        return reportingTokens[finalPayoutDistributionHash];
+    }
+
+    function getShareToken(uint8 outcome)  public constant returns (IShareToken) {
+        require(outcome < numOutcomes);
+        return shareTokens[outcome];
+    }
+
+    function getFinalPayoutDistributionHash() public constant returns (bytes32) {
+        return finalPayoutDistributionHash;
+    }
+
+    function getPayoutDenominator() public constant returns (uint256) {
+        return payoutDenominator;
+    }
+
+    function getDenominationToken() public constant returns (Cash) {
+        return cash;
+    }
+
+    function getMarketCreatorSettlementFeeInAttoethPerEth() public constant returns (uint256) {
+        return feePerEthInAttoeth;
+    }
+
+    function getMaxDisplayPrice() public constant returns (int256) {
+        return maxDisplayPrice;
+    }
+
+    function getMinDisplayPrice() public constant returns (int256) {
+        return minDisplayPrice;
+    }
+
+    function getCompleteSetCostInAttotokens() public constant returns (uint256) {
+        return uint256(maxDisplayPrice.sub(minDisplayPrice));
+    }
+
+    function getTopic() public constant returns (bytes32) {
+        return topic;
+    }
+
+    function shouldCollectReportingFees() public constant returns (bool) {
+        return !getBranch().isContainerForShareToken(cash);
+    }
+
+    function isDoneWithAutomatedReporters() public constant returns (bool) {
+        return automatedReportReceived || block.timestamp > getAutomatedReportDueTimestamp();
+    }
+
+    function isDoneWithLimitedReporters() public constant returns (bool) {
+        return getReportingState() > ReportingState.LIMITED_REPORTING;
+    }
+
+    function isDoneWithAllReporters() public constant returns (bool) {
+        return getReportingState() > ReportingState.ALL_REPORTING;
+    }
+
+    function isFinalized() public constant returns (bool) {
```

Comment:
For consistency I think I'll move the logic into `getReportingState` but keep this function and make it a convenience for the reason you stated.